### ISSUE TITLE
purge execution logs "Execution History Clean" new feature don't work.

### ISSUE THREAD

**Issue by philjov** [rundeck/rundeck#5274]
**Describe the bug**
purge execution logs "Execution History Clean" new feature don't work.

**My Rundeck detail**

* Rundeck version: 3.10
* install type: rpm
* OS Name/version: Oracle Linux 7.2
* DB Type/version: postgres 9.6.11 (AWS RDS)

**To Reproduce**
1. Select a project with daily activity / go to project setting - edit configuration - Execution History Clean / Set Up Day to keep - minimum exec - max size of deletion - schedule 
2. Check in job activity
- number of executions > parameter set in Execution History Clean
- oldest execution  > parameter set in Execution History Clean
3. Check logs : no logs available?
4. new feature don't work ?

**Expected behavior**
Execution history clean ok

**Desktop (please complete the following information):**
 - Browser : chrome

**Comment by MegaDrive68k** [comment]
Confirmed.

With this config (clean executions each minute keeping 7 executions):

![Screenshot_1](https://user-images.githubusercontent.com/51376003/65255480-5bd33100-dad4-11e9-8b2b-9939239919cc.png)

But the executions keep on Activity:

![Screenshot_2](https://user-images.githubusercontent.com/51376003/65255501-64c40280-dad4-11e9-9954-2a872d83a242.png)

Also, at the moment of delete some execution the log show this (but the execution is deleted):

`[2019-09-19 11:11:53.092]  WARN ExecutionService --- [p1584119395-153] Failed to delete file while deleting execution 173: /home/user/Rundeck/3.1.0/var/logs/rundeck/ProjectLOCAL/job/786ec8c6-7e3c-4082-96d8-b3a3ce611b46/logs/173.state.json`

Thanks for your feedback @philjov!

**Comment by heronimus** [comment]
confirmed above problem.
also when using rd-cli to bulk delete execution it show below WARN log on service.log 

`[2019-09-27 18:32:40.544]  WARN ExecutionService --- [848495371-57012] Failed to delete file while deleting execution 30988914: /var/lib/rundeck/logs/rundeck/SCHEDULER-ADMIN/job/af96e42b-2ec4-45be-9bc5-ebced4b069ac/logs/30988914.rdlog
[2019-09-27 18:32:40.547]  WARN ExecutionService --- [848495371-57012] Failed to delete file while deleting execution 30988914: /var/lib/rundeck/logs/rundeck/SCHEDULER-ADMIN/job/af96e42b-2ec4-45be-9bc5-ebced4b069ac/logs/30988914.state.json
[2019-09-27 18:32:40.692]  WARN ExecutionService --- [848495371-57012] Failed to delete file while deleting execution 30988910: /var/lib/rundeck/logs/rundeck/SCHEDULER-ADMIN/job/af96e42b-2ec4-45be-9bc5-ebced4b069ac/logs/30988910.rdlog
[2019-09-27 18:32:40.696]  WARN ExecutionService --- [848495371-57012] Failed to delete file while deleting execution 30988910: /var/lib/rundeck/logs/rundeck/SCHEDULER-ADMIN/job/af96e42b-2ec4-45be-9bc5-ebced4b069ac/logs/30988910.state.json`

**Comment by MegaDrive68k** [comment]
Thanks for your feedback @heronimus 

Confirmed on 3.1.1 too.

Regards.

**Comment by rdoering** [comment]
Im on Rundeck 3.1.2-20190927 and in my case the *..state.json and *.rdlog are removed but the *.execution.xml are still there :-/.

@gschueler Will this be fixed in v3.2.0 also?

**Comment by gschueler** [comment]
@rdoering yes, if you want to verify it you can try the snapshot

**Comment by reziel84** [comment]
i have the same issue on rundeck 3.1.2 on centos 7, i try to just launch yum update rundeck but yum found only 3.1.3 version and not 3.2 version, in which way i can update to 3.2?

**Comment by tsde** [comment]
@reziel84 Version 3.2 is not out yet. The easiest way to try it before it gets released is using the official docker image with then SNAPSHOT tag.

@gschueler I was wondering what kind of behavior has been implemented for this feature in a clustered environment. Say I have two Rundeck instances. As this feature is set project-wide, may the two instances try to trigger the cleanup at the same time or is there some kind of specific implementation (a lock, a "leader" declared for this action in the same way as the scheduled jobs feature, etc...) which prevent it from happening ?

**Comment by wbob** [comment]
I stumbled on this in a 3.1.2 install, updated to 3.2, raised the max deletion number to catch up in history. Though some loot was left over and generated errors. To get rid of xml files without reference in the database, I ran
```
#!/usr/bin/env bash
# example: /var/lib/rundeck/logs/rundeck/project/job/0a071e24-d4a6-463f-a6fe-367d74556408/logs/86699.execution.xml
# check execution table for lowest id: 
id=10000
for xml in $(find /var/lib/rundeck/logs/rundeck/project/job/ -name '*.execution.xml'); do
  path=${xml%%\.execution\.xml}
  num=${path##*logs\/}
  if [ $num -lt $id ]; then
    rm -v "$xml"
  fi
done
```

**Comment by wadelu** [comment]
Was this fixed? We faced same issue on 3.3.10
